libraries/Microsoft.Bot.Connector/Authentication/ICredentialProvider.cs

```diff
@@ -17,6 +17,22 @@ namespace Microsoft.Bot.Connector.Authentication
     /// </summary>
     public interface ICredentialProvider
     {
+        /// <summary>
+        /// Gets or sets the app ID for this credential.
+        /// </summary>
+        /// <value>
+        /// The app ID for this credential.
+        /// </value>
+        string AppId { get; set; }
```

Comment:
Unfortunately, adding methods to an interface is a breaking change. Anyone who has implemented this will have broken code that will not compile. This solution won't work either. :(

Reply:
Hi @cleemullins, 

We refactored the code to avoid the breaking change. 
Now, we added a new method which checks through reflection, if any object implementing _ICredentialProvider_ has an AppID. In case this property is null, a NullArgumentException will be thrown, as is a required parameter.

Thanks!